server/routerlicious/packages/services-shared/src/socketIoServer.ts

```diff
@@ -260,6 +313,58 @@ class SocketIoServer implements core.IWebSocketServer {
 			this.redisClientConnectionManagerForPub.getRedisClient().quit(),
 			this.redisClientConnectionManagerForSub.getRedisClient().quit(),
 		]);
+		if (this.socketIoConfig?.pingPongLatencyTrackingEnabled) {
+			clearInterval(this.pingPongLatencyInterval);
+			this.flushPingPongLatencyTracking();
+		}
+	}
+
+	private initPingPongLatencyTracking(socket: Socket) {
+		if (!this.socketIoConfig?.pingPongLatencyTrackingEnabled) {
+			return;
+		}
+
+		let lastPingStartTime: number | undefined;
+		const packetCreateHandler = (packet: any) => {
+			if (packet.type === "ping") {
+				lastPingStartTime = Date.now();
+			}
+		};
+		socket.conn.on("packetCreate", packetCreateHandler);
+		const packetReceivedHandler = (packet: any) => {
+			if (packet.type === "pong") {
+				if (lastPingStartTime !== undefined) {
+					this.apiCounters.incrementCounter("pingPongCount", 1);
+					this.apiCounters.incrementCounter(
+						"pingPongLatency",
+						Date.now() - lastPingStartTime,
```

Comment:
Eventually I guess you can only use avg of the latency by doing `Latency/count`?

Reply:
Right, yes. It feels like a good compromise compared to generating a 1 log per socket per 20s, which I think would add up way too quickly.

Always happy to revisit that assumption.